hs_core/views/__init__.py

```diff
@@ -1719,6 +1707,41 @@ def _set_resource_sharing_status(user, resource, flag_to_set, flag_value):
     return None
 
 
+class FindGroupsView(TemplateView):
+    template_name = 'pages/groups-unauthenticated.html'  # default view is for users not logged in
+
+    def dispatch(self, *args, **kwargs):
+        if self.request.user.is_authenticated():
+            self.template_name = 'pages/groups-authenticated.html'  # update template if user is logged in
+        return super(FindGroupsView, self).dispatch(*args, **kwargs)
+
+    def get_context_data(self, **kwargs):
+        if self.request.user.is_authenticated():
+            u = User.objects.get(pk=self.request.user.id)
+
+            groups = Group.objects.filter(gaccess__active=True).filter(gaccess__unlisted=False).exclude(name="Hydroshare Author")
+            for g in groups:
+                g.is_user_member = u in g.gaccess.members
+                g.join_request_waiting_owner_action = g.gaccess.group_membership_requests.filter(
+                    request_from=u).exists()
+                g.join_request_waiting_user_action = g.gaccess.group_membership_requests.filter(
+                    invitation_to=u).exists()
+                g.join_request = None
+                if g.join_request_waiting_owner_action or g.join_request_waiting_user_action:
+                    g.join_request = g.gaccess.group_membership_requests.filter(request_from=u).first() or \
+                                     g.gaccess.group_membership_requests.filter(invitation_to=u).first()
+            return {
+                'profile_user': u,
+                'groups': groups
+            }
+        else:
+            groups = GroupAccess.groups_with_public_resources().exclude(name="Hydroshare Author")  # active is included in this query
```

Comment:
Why only list groups with public resources?  Shouldn't it list all listed groups just like when you're logged in?  And why are both queries excluding Hydroshare Author?

Reply:
Let's get on a call for this. I'll wait until Alva is back, so we can all be on the same page.

Reply:
The point of the exhibit page is to exhibit group products. I specifically designed it to ignore groups with no products, so that people who click on a group will always see "something".

Reply:
I was not aware of this type of behavior, but it sounds like this is functionality that was asked for and it behaves as expected.